### PR TITLE
fix(validator_client): raise soft fd limit

### PR DESCRIPTION
**Pull request by rkrasiuk**
## Issue Addressed

Closes https://github.com/sigp/lighthouse/issues/3468

## Proposed Changes

Attempt to raise validator client's fdlimit. OS specific behavior is described in the comments

### CODE REVIEW

File: validator_client/Cargo.toml [hunk 1]
```diff
@@ -62,3 +62,4 @@ malloc_utils = { workspace = true }
 sysinfo = { workspace = true }
 system_health = { path = "../common/system_health" }
 logging = { workspace = true }
+fdlimit = "0.2.1"
```
Comment (dapplion) [inline validator_client/Cargo.toml]: Can you use version 0.3.0? It includes an improvement to not panic on syscall errors, https://github.com/paritytech/fdlimit/commit/eee618fbe779452099f1857dd9191ec711067af6

**Review by dapplion — DISMISSED** [review]
I have tested the fslimit crate on a separate repo on my local M2 mac on a regular user and it could raise the fd limit from 256 -> 10240.

However I have mixed feelings about this feature. It seems to only be useful to large operators or devnet operators, two parties who can manually raise the fd limit on their own.

I would recommend mantainers to take a look at the fdlimit crate briefly since it's just one short file https://github.com/paritytech/fdlimit/blob/master/src/lib.rs

To scope its usage better the validator client could only try to raise the fdlimit if needs too i.e. if the user is loading more than N keys.

**Comment by rkrasiuk** [comment]
@dapplion upgraded the crate. re limiting this behavior - fair enough, we could raise fdlimit based on the current limit and the number of user keys loaded. alternatively, we could expose a CLI flag, that users could pass in case they hit the limit

**Comment by dapplion** [comment]
> alternatively, we could expose a CLI flag, that users could pass in case they hit the limit

This feature is maximally useful when it allows users to not have to think about file limits. Adding a flag defeats the purpose, as after an error the user now has to add a new flag, might as well manually bump the file limit

**Comment by paulhauner** [comment]
I've pushed this to v5.1.0 since it doesn't seem critical for Deneb.

**Comment by pawanjay176** [comment]
Pushed back to 5.2 since we want to get networking improvements in the 5.1 release out soon.

**Review by dapplion — APPROVED** [review]

**Comment by realbigsean** [comment]
@mergify queue

**Pull request merged**